templating/base_render_options.go

```diff
@@ -123,24 +123,80 @@ func OverrideValuesWithBuildInfo(c1 *Config, c2 *Config, opts *BaseRenderOptions
 	return base, nil
 }
 
-// LoadAndRenderSteps loads a template file and renders it according to an optional values file, --set values,
+// LoadAndRenderBuildSteps loads a template file for build and renders it according to an optional values file, --set values,
+// and base render options.
+func LoadAndRenderBuildSteps(ctx context.Context, template *Template, opts *BaseRenderOptions) (string, error) {
+	// load steps and override values
+	mergedVals, err := LoadSteps(template, opts)
+	if err != nil {
+		return "", fmt.Errorf("error while loading build steps: %v", err)
+	}
+
+	engine := NewEngine()
+
+	rendered, err := engine.Render(template, mergedVals)
+	if err != nil {
+		return "", fmt.Errorf("error while rendering templates: %v", err)
+	}
+
+	if rendered == "" {
+		return "", errors.New("rendered template was empty")
+	}
+
+	return rendered, nil
+}
+
+// LoadAndRenderSteps loads a template file for exec and renders it according to an optional values file, --set values,
 // and base render options.
 func LoadAndRenderSteps(ctx context.Context, template *Template, opts *BaseRenderOptions) (string, error) {
+	// load steps and override values
+	mergedVals, err := LoadSteps(template, opts)
+	if err != nil {
+		return "", fmt.Errorf("error while loading exec steps: %v", err)
+	}
 	// return empty rendered string for an empty template.
-	if len(template.GetData()) == 0 {
+	if mergedVals == nil {
 		return "", nil
 	}
 
+	engine := NewEngine()
+	// we will pass nil for the secret resolve override so as to use the default resolve function.
+	secrets, err := renderAndResolveSecrets(ctx, template, engine, nil, opts, mergedVals)
+	if err != nil {
+		return "", fmt.Errorf("failed to resolve secrets in the task with error: %v", err)
+	}
+	// update the secrets collection with resolved secrets.
+	mergedVals["Secrets"] = secrets
+
+	rendered, err := engine.Render(template, mergedVals)
+	if err != nil {
+		return "", fmt.Errorf("error while rendering templates: %v", err)
+	}
+
+	if rendered == "" {
+		return "", errors.New("rendered template was empty")
+	}
+
+	return rendered, nil
+}
+
+// LoadSteps loads a template file and overrides values with build info
+func LoadSteps(template *Template, opts *BaseRenderOptions) (Values, error) {
```

Comment:
>LoadSteps [](start = 5, length = 9)

if you don't plan to expose outside the module, lowercase loadSteps to make it private

Reply:
That's a cool go feature. Yes, this method should be private